From 1d84c22352954ae9628cd73e54b308e41035fce9 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Wed, 18 Feb 2026 14:41:48 +0800
Subject: [PATCH 451/451] Revert "AOSCOS: drm: amdgpu: radeon: disable cache
 flush workaround for LoongArch and Loongson64"

This reverts commit 894addd0e3567428aa2070be4f513f1c40265a86.
---
 drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c | 8 --------
 drivers/gpu/drm/amd/amdgpu/gfx_v8_0.c | 8 --------
 drivers/gpu/drm/radeon/cik.c          | 8 --------
 3 files changed, 24 deletions(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c b/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
index d455a752f3ff1..2b7aba22ecc19 100644
--- a/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gfx_v7_0.c
@@ -2125,13 +2125,6 @@ static void gfx_v7_0_ring_emit_fence_gfx(struct amdgpu_ring *ring, u64 addr,
 	bool int_sel = flags & AMDGPU_FENCE_FLAG_INT;
 	bool exec = flags & AMDGPU_FENCE_FLAG_EXEC;
 
-/* This workaround causes instability for LoongArch/Loongson (MIPS)
- * devices based on the 7A1000/2000 chipset under heavy I/O load.
- *
- * FIXME: Disable this workaround until we find a better fix (possibly in
- * the platform-specific PCI code).
- */
-#ifndef CONFIG_MACH_LOONGSON64
 	/* Workaround for cache flush problems. First send a dummy EOP
 	 * event down the pipe with seq one below.
 	 */
@@ -2145,7 +2138,6 @@ static void gfx_v7_0_ring_emit_fence_gfx(struct amdgpu_ring *ring, u64 addr,
 				DATA_SEL(1) | INT_SEL(0));
 	amdgpu_ring_write(ring, lower_32_bits(seq - 1));
 	amdgpu_ring_write(ring, upper_32_bits(seq - 1));
-#endif
 
 	/* Then send the real EOP event down the pipe. */
 	amdgpu_ring_write(ring, PACKET3(PACKET3_EVENT_WRITE_EOP, 4));
diff --git a/drivers/gpu/drm/amd/amdgpu/gfx_v8_0.c b/drivers/gpu/drm/amd/amdgpu/gfx_v8_0.c
index 61b999f2e8c80..8a81713d97aac 100644
--- a/drivers/gpu/drm/amd/amdgpu/gfx_v8_0.c
+++ b/drivers/gpu/drm/amd/amdgpu/gfx_v8_0.c
@@ -6113,13 +6113,6 @@ static void gfx_v8_0_ring_emit_fence_gfx(struct amdgpu_ring *ring, u64 addr,
 	bool int_sel = flags & AMDGPU_FENCE_FLAG_INT;
 	bool exec = flags & AMDGPU_FENCE_FLAG_EXEC;
 
-/* This workaround causes instability for LoongArch/Loongson (MIPS)
- * devices based on the 7A1000/2000 chipset under heavy I/O load.
- *
- * FIXME: Disable this workaround until we find a better fix (possibly in
- * the platform-specific PCI code).
- */
-#ifndef CONFIG_MACH_LOONGSON64
 	/* Workaround for cache flush problems. First send a dummy EOP
 	 * event down the pipe with seq one below.
 	 */
@@ -6134,7 +6127,6 @@ static void gfx_v8_0_ring_emit_fence_gfx(struct amdgpu_ring *ring, u64 addr,
 				DATA_SEL(1) | INT_SEL(0));
 	amdgpu_ring_write(ring, lower_32_bits(seq - 1));
 	amdgpu_ring_write(ring, upper_32_bits(seq - 1));
-#endif
 
 	/* Then send the real EOP event down the pipe:
 	 * EVENT_WRITE_EOP - flush caches, send int */
diff --git a/drivers/gpu/drm/radeon/cik.c b/drivers/gpu/drm/radeon/cik.c
index 170d54c4b5eb5..e7b2108770dc9 100644
--- a/drivers/gpu/drm/radeon/cik.c
+++ b/drivers/gpu/drm/radeon/cik.c
@@ -3543,13 +3543,6 @@ void cik_fence_gfx_ring_emit(struct radeon_device *rdev,
 	struct radeon_ring *ring = &rdev->ring[fence->ring];
 	u64 addr = rdev->fence_drv[fence->ring].gpu_addr;
 
-/* This workaround causes instability for LoongArch/Loongson (MIPS)
- * devices based on the 7A1000/2000 chipset under heavy I/O load.
- *
- * FIXME: Disable this workaround until we find a better fix (possibly in
- * the platform-specific PCI code).
- */
-#ifndef CONFIG_MACH_LOONGSON64
 	/* Workaround for cache flush problems. First send a dummy EOP
 	 * event down the pipe with seq one below.
 	 */
@@ -3563,7 +3556,6 @@ void cik_fence_gfx_ring_emit(struct radeon_device *rdev,
 				DATA_SEL(1) | INT_SEL(0));
 	radeon_ring_write(ring, fence->seq - 1);
 	radeon_ring_write(ring, 0);
-#endif
 
 	/* Then send the real EOP event down the pipe. */
 	radeon_ring_write(ring, PACKET3(PACKET3_EVENT_WRITE_EOP, 4));
-- 
2.52.0

